Gem/Code/Source/Weapons/TraceWeapon.cpp

```diff
@@ -33,7 +33,14 @@ namespace MultiplayerSample
             if (isMultiSegmented)
             {
                 ActiveShot activeShot{ eventData.m_initialTransform, eventData.m_targetPosition, LifetimeSec{ 0.0f } };
-                weaponState.m_activeShots.emplace_back(activeShot);
+                if (weaponState.m_activeShots.size() < weaponState.m_activeShots.max_size())
+                { 
+                    weaponState.m_activeShots.emplace_back(activeShot);
+                }
+                else
+                {
+                    AZ_Assert(false, "Attempting to add too many active shots to the TraceWeapon.");
```

Comment:
will this assert show up in the Game.log? (admittedly i'm not super clear on which types of logs end up there)

Reply:
I'm not 100% sure right now with our current settings. We _want_ asserts to show up in our logs, but I haven't been able to validate that they actually are yet.

Reply:
Yep, I can verify that they showed up in the Game.log:

```
<16:42:17> (PerformanceCollector) - Waited 0 seconds. Will start collecting performance numbers for 2000 frames at batch 99998...
<16:42:50> 
==================================================================
<16:42:50> Trace::Assert
 D:/github/o3de-multiplayersample/Gem/Code/Source/Weapons/TraceWeapon.cpp(44): (2452) 'void __cdecl MultiplayerSample::TraceWeapon::Activate(struct MultiplayerSample::WeaponState &,const class Multiplayer::ConstNetworkEntityHandle,struct MultiplayerSample::ActivateEvent &,bool)'
<16:42:50> Attempting to add too many active shots to the TraceWeapon.
<16:42:50> ------------------------------------------------
<16:42:51> D:\github\o3de-multiplayersample\Gem\Code\Source\Weapons\TraceWeapon.cpp (46) : MultiplayerSample::TraceWeapon::Activate
<16:42:51> D:\github\o3de-multiplayersample\Gem\Code\Source\Components\NetworkWeaponsComponent.cpp (112) : MultiplayerSample::NetworkWeaponsComponent::ActivateWeaponWithParams
<16:42:51> D:\github\o3de\Code\Framework\AzCore\AzCore\std\function\function_template.h (172) : AZStd::Internal::function_util::get_invoker<void __cdecl(int,unsigned char),AZStd::Internal::function_util::function_obj_tag,AZStd::allocator>::call<`MultiplayerSample::
<16:42:51> D:\github\o3de\Code\Framework\AzCore\AzCore\EBus\Event.inl (288) : AZ::Event<int,unsigned char>::Signal
<16:42:51> D:\github\o3de-multiplayersample\build\windows_mono\External\o3de-multiplayersample-4a925c78\Gem\Code\Azcg\Generated\MultiplayerSample.Client.Static\Source\AutoGen\NetworkWeaponsComponent.AutoComponent.cpp (792) : MultiplayerSample::NetworkWeaponsCompone
<16:42:51> D:\github\o3de\Gems\Multiplayer\Code\Source\Components\NetBindComponent.cpp (493) : Multiplayer::NetBindComponent::HandlePropertyChangeMessage
<16:42:51> D:\github\o3de\Gems\Multiplayer\Code\Source\NetworkEntity\EntityReplication\EntityReplicationManager.cpp (709) : Multiplayer::EntityReplicationManager::HandlePropertyChangeMessage
<16:42:51> D:\github\o3de\Gems\Multiplayer\Code\Source\NetworkEntity\EntityReplication\EntityReplicationManager.cpp (894) : Multiplayer::EntityReplicationManager::HandleEntityUpdateMessage
<16:42:51> D:\github\o3de-multiplayersample\build\windows_mono\External\Multiplayer-cb8b491b\Code\Azcg\Generated\Multiplayer.Common.Static\Source\AutoGen\Multiplayer.AutoPacketDispatcher.inl (134) : MultiplayerPackets::DispatchPacket<Multiplayer::MultiplayerSystemC
<16:42:51> D:\github\o3de\Code\Framework\AzNetworking\AzNetworking\UdpTransport\UdpFragmentQueue.cpp (170) : AzNetworking::UdpFragmentQueue::ProcessReceivedChunk
<16:42:51> D:\github\o3de\Code\Framework\AzNetworking\AzNetworking\UdpTransport\UdpConnection.cpp (305) : AzNetworking::UdpConnection::HandleCorePacket
<16:42:51> D:\github\o3de\Code\Framework\AzNetworking\AzNetworking\UdpTransport\UdpNetworkInterface.cpp (294) : AzNetworking::UdpNetworkInterface::Update
<16:42:51> D:\github\o3de\Code\Framework\AzNetworking\AzNetworking\Framework\NetworkingSystemComponent.cpp (79) : AzNetworking::NetworkingSystemComponent::OnSystemTick
<16:42:51> D:\github\o3de\Code\Framework\AzCore\AzCore\EBus\Internal\BusContainer.h (1360) : AZ::Internal::EBusContainer<AZ::SystemTickEvents,AZ::SystemTickEvents,0,1>::Dispatcher<AZ::EBus<AZ::SystemTickEvents,AZ::SystemTickEvents> >::Broadcast<void (__cdecl AZ::Sy
<16:42:51> D:\github\o3de\Code\Framework\AzCore\AzCore\Component\ComponentApplication.cpp (1506) : AZ::ComponentApplication::TickSystem
<16:42:51> D:\github\o3de\Code\LauncherUnified\Launcher.cpp (571) : O3DELauncher::Run
<16:42:51> D:\github\o3de\Code\LauncherUnified\Platform\Windows\Launcher_Windows.cpp (29) : WinMain
<16:42:51> D:\a\_work\1\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl (288) : __scrt_common_main_seh
<16:42:51> 00007FFD18397614 (KERNEL32) : BaseThreadInitThunk
<16:42:51> 00007FFD187626A1 (ntdll) : RtlUserThreadStart
<16:42:51> ==================================================================
<16:42:51> ====Assert added to ignore list by spec and verbosity setting.====
<16:42:51> [Warning] (System) - Processing time exceeded, discarding 24/35 received packets
```